### PR TITLE
Add RDP cache use-case files

### PR DESCRIPTION
**Pull request by efelmandar**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-2714)

## Description
Adding all the necessary files for the RDP Cache Hunting use case pack which includes Playbooks, Automation scripts, Layout, Incident Types and Incident fields

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [x] 6.8.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Documentation

### CODE REVIEW

File: Packs/CommonScripts/Scripts/BMCTool/BMCTool.py [hunk 1]
```diff
@@ -0,0 +1,502 @@
...
+            self.b_log("debug", True, "Old data will also be saved in separate files.")
+
+    def b_log(self, ltype, verbose, lmsg):
+
+        log_type = {"info": demisto.info,
+                    "debug": demisto.debug,
+                    "error": demisto.error
+                    }
+
+        if not verbose or self.verb:
```
Comment (eyalpalo) [inline Packs/CommonScripts/Scripts/BMCTool/BMCTool.py:90]: Don't really understand the condition, if verbose is set to False we will print the log.
Why not 
`if verbose or self.verb`?

File: Packs/RDPCacheHunting/Scripts/SetRDPOverallScore/SetRDPOverallScore.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        'ContentsFormat': formats['html'],
+        'Type': entryTypes['note'],
+        'Contents': html
+    })
+
+
+if __name__ in ["__main__", "builtin", "builtins"]:
+    main()
+
+# register_module_line('RapidBreachResponse-CompletedTasksCount-Widget', 'end', __line__())
```
Comment (eyalpalo) [inline Packs/RDPCacheHunting/Scripts/SetRDPOverallScore/SetRDPOverallScore.py:38]: Please remove this line

File: Packs/RDPCacheHunting/Scripts/SetRDPOverallScore/SetRDPOverallScore.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+commonfields:
+  id: SetRDPOverallScore
+  version: -1
+name: SetRDPOverallScore
```
Comment (eyalpalo) [inline Packs/RDPCacheHunting/Scripts/SetRDPOverallScore/SetRDPOverallScore.yml:4]: Same note regarding the outputs as above.

File: Packs/CommonScripts/Scripts/BMCTool/BMCTool.py [hunk 4]
```diff
@@ -0,0 +1,502 @@
...
+                            self.b_log("error", False,
+                                       "Uncompressed tile data seems bogus (uncompressed %d bytes while expecting %d). Discarding tile." % (
+                                           len(t_bmp), t_width * t_height * bl // (64 * 64)))
+                            t_bmp = b""
+                        else:
+                            t_bmp = self.b_parse_rgb565(t_bmp)
+                else:
+                    cf = t_len // (t_width * t_height)
+                    if cf == 4:
+                        t_bmp = self.b_parse_rgb32b(self.bdat[len(t_hdr):len(t_hdr) + cf * t_width * t_height])
```
[the remark this answers is not in the document]
Comment (eyalpalo) [reply Packs/CommonScripts/Scripts/BMCTool/BMCTool.py:174]: For example:
image

**Review by eyalpalo — CHANGES_REQUESTED** [review]
Almost there! looks good
Few more changes. Lets have a final demo (maybe worth also inviting Yarden?)

File: Packs/CommonScripts/Scripts/StringSimilarity/README.md
Comment (melamedbn) [inline Packs/CommonScripts/Scripts/StringSimilarity/README.md:1]: Should it be empty?

File: Packs/CommonScripts/Scripts/StringSimilarity/StringSimilarity_test.py
Comment (melamedbn) [inline Packs/CommonScripts/Scripts/StringSimilarity/StringSimilarity_test.py:1]: Should it be empty?

File: Packs/RDPCacheHunting/IncidentTypes/incidenttype-RDP_Sessions.json
Comment (melamedbn) [inline Packs/RDPCacheHunting/IncidentTypes/incidenttype-RDP_Sessions.json:1]: Maybe we should reconsider the name? the type is "RDP Sessions Hunting" or something like that

File: Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml [hunk 5]
```diff
@@ -0,0 +1,1525 @@
...
+  description: When set to True, will use XDR to get RDP cache files from the endpoints, but set to False will try and use existing cache file from context
+  playbookInputQuery:
+- key: EndpointIDs
+  value: {}
+  required: false
+  description: A comma seperated list of endpoint ID's to retrieve cache files from
+  playbookInputQuery:
+- key: FilePath
+  value:
+    simple: C:\Users\administrator\AppData\Local\Microsoft\Terminal Server Client\Cache\
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml:1510]: If folder path is eligible, let's document it and give another example in the description.

File: Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml [hunk 6]
```diff
@@ -0,0 +1,1525 @@
...
+      "dimensions": {
+        "height": 3460,
+        "width": 2530,
+        "x": -840,
+        "y": -1365
+      }
+    }
+  }
+inputs:
+- key: ShouldCollectRDPCache
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml:1497]: I see you also used the Retrieve File from Endpoint - Generic V3 playbook.
That means that we need to document in the playbook's readme and the input description that we can collect using x,y,z.

File: Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml [hunk 7]
```diff
@@ -0,0 +1,1525 @@
+id: RDP Bitmap Cache - Detect and Hunt
+version: -1
+contentitemexportablefields:
+  contentitemfields: {}
+name: RDP Bitmap Cache - Detect and Hunt
+description: This playbook automates the collection and forensic analysis of RDP sessions cache data by collecting the cache files and convert it to an image, extract readable text from the image and build IOC's from text and finally enrich any extracted indicators for further hunting
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml:6]: Use MD instead.
I see you also used the Retrieve File from Endpoint - Generic V3 playbook.
That means that we need to document in the playbook's readme that we can collect using x,y,z.

File: Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml [hunk 8]
```diff
@@ -0,0 +1,1525 @@
...
+  "36":
+    id: "36"
+    taskid: d6dae20c-ec34-4bdf-8506-1783736ceb2f
+    type: regular
+    task:
+      id: d6dae20c-ec34-4bdf-8506-1783736ceb2f
+      version: -1
+      name: Get RDP cache files
+      description: Retrieves files from selected endpoints. You can retrieve up to 20 files, from no more than 10 endpoints. At least one endpoint ID and one file path are necessary in order to run the command. After running this command, you can use the xdr-action-status-get command with returned action_id, to check the action status.
+      script: '|||xdr-file-retrieve'
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml:1014]: Change to the matching Core command.

File: Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml [hunk 9]
```diff
@@ -0,0 +1,1525 @@
...
+      name: Get Cache Files
+      type: title
+      iscommand: false
+      brand: ""
+      description: ''
+    nexttasks:
+      '#none#':
+      - "36"
+      - "34"
+      - "31"
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml:988]: Please remove

File: Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml [hunk 10]
```diff
@@ -0,0 +1,1525 @@
...
+          root: File
+          filters:
+          - - operator: containsString
+              left:
+                value:
+                  simple: File.Extension
+                iscontext: true
+              right:
+                value:
+                  simple: bin
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml:119]: Why are you checking for bin files only?

File: Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml [hunk 11]
```diff
@@ -0,0 +1,1525 @@
...
+          right:
+            value:
+              simple: bmc
+          ignorecase: true
+        - operator: isExists
+          left:
+            value:
+              complex:
+                root: incident
+                accessor: attachment
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml:829]: Add filter for the attachment extension.

File: Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml [hunk 12]
```diff
@@ -0,0 +1,1525 @@
...
+    isautoswitchedtoquietmode: false
+  "31":
+    id: "31"
+    taskid: c5d895cc-0d8a-49b2-824a-449eff38bd5f
+    type: condition
+    task:
+      id: c5d895cc-0d8a-49b2-824a-449eff38bd5f
+      version: -1
+      name: Was the file retrieved successfully?
+      description: Checkif the file was successfully retrieved
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml:789]: Add space Checkif -> Check if

File: Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml [hunk 13]
```diff
@@ -0,0 +1,1525 @@
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "47":
+    id: "47"
+    taskid: baef2a3c-06f5-454a-87d1-ce1a06e0b83f
+    type: regular
+    task:
+      id: baef2a3c-06f5-454a-87d1-ce1a06e0b83f
+      version: -1
+      name: Convert pdf to image
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml:1428]: Change to "Convert image to pdf"

File: Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml [hunk 14]
```diff
@@ -0,0 +1,1525 @@
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "45":
+    id: "45"
+    taskid: 2c88b6f8-a009-4f12-8088-5361c0f64338
+    type: regular
+    task:
+      id: 2c88b6f8-a009-4f12-8088-5361c0f64338
+      version: -1
+      name: Convert image to png
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml:1308]: Convert the bmp file to png

File: Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml [hunk 15]
```diff
@@ -0,0 +1,1525 @@
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "46":
+    id: "46"
+    taskid: 5b45875f-222b-4951-8800-8bfa8a377210
+    type: regular
+    task:
+      id: 5b45875f-222b-4951-8800-8bfa8a377210
+      version: -1
+      name: Convert image to pdf
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml:1368]: Rasterize the PDF file

File: Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml [hunk 16]
```diff
@@ -0,0 +1,1525 @@
...
+          root: File
+          filters:
+          - - operator: startWith
+              left:
+                value:
+                  simple: File.Name
+                iscontext: true
+              right:
+                value:
+                  simple: original
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml:67]: We want to OCR both sharpened and greyscale images, the original is only for display.

File: Packs/RDPCacheHunting/doc_files/Set_list_of_indicator_types.png
Comment (melamedbn) [inline Packs/RDPCacheHunting/doc_files/Set_list_of_indicator_types.png:1]: Can we remove it?

File: Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml [hunk 17]
```diff
@@ -0,0 +1,1525 @@
...
+          - operator: uniq
+          - operator: split
+            args:
+              delimiter:
+                value:
+                  simple: ' '
+      string_B:
+        complex:
+          root: lists
+          accessor: MalTools
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml:171]: If we won't provide the list we should retrieve it somehow.

File: Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml [hunk 18]
```diff
@@ -0,0 +1,1525 @@
...
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "26":
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml:561]: Add the splunk and qradar time ranges to the playbook inputs.

File: Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml [hunk 19]
```diff
@@ -0,0 +1,1525 @@
...
+          root: File.Name
+          filters:
+          - - operator: startWith
+              left:
+                value:
+                  simple: File.Name
+                iscontext: true
+              right:
+                value:
+                  simple: original
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml:223]: Let's use the greyscale and sharpened.

File: Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml [hunk 20]
```diff
@@ -0,0 +1,1525 @@
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "7":
+    id: "7"
+    taskid: 421a0baf-99f2-4d53-8765-896cbe4b1242
+    type: regular
+    task:
+      id: 421a0baf-99f2-4d53-8765-896cbe4b1242
+      version: -1
+      name: Suspicious strings hunting
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml:202]: Add the min_score and limit as a playbook inputs.

File: Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml [hunk 21]
```diff
@@ -0,0 +1,1525 @@
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "27":
+    id: "27"
+    taskid: bcc6d2ba-4b32-4c3c-87ba-c4d780443c77
+    type: regular
+    task:
+      id: bcc6d2ba-4b32-4c3c-87ba-c4d780443c77
+      version: -1
+      name: Hunt RDP IoCs - Exe + Path
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml:630]: Why do we use only splunk?
We should make it generic.

File: Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml [hunk 22]
```diff
@@ -0,0 +1,1525 @@
...
+      description: Searches Splunk for events.
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml:631]: Please elaborate.

File: Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml [hunk 23]
```diff
@@ -0,0 +1,1525 @@
...
+      description: This automation calculates the similarity ratio between a Text and a list of strings and outputs a decimal value between 0.0 and 1.0 (1.0 if the sequences are identical, and 0.0 if they don't have anything in common).
+      scriptName: StringSimilarity
+      type: regular
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "15"
+    scriptarguments:
+      similiarity_threshold:
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml:155]: Add the similarity_treshold as a playbook input.

File: Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml [hunk 24]
```diff
@@ -0,0 +1,1525 @@
...
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "29":
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml:709]: Check if we can remove the sub playbook and put the tasks in the main one.

File: Packs/RDPCacheHunting/Layouts/layoutscontainer-RDP_Sessions.json [hunk 25]
```diff
@@ -0,0 +1,1311 @@
...
+                                "id": "4dc32de0-18dd-11ee-8250-4d18bfc11034",
+                                "index": 1,
+                                "sectionItemType": "field",
+                                "startCol": 0
+                            }
+                        ],
+                        "maxW": 3,
+                        "minH": 1,
+                        "moved": false,
+                        "name": "Strings Similarity",
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Layouts/layoutscontainer-RDP_Sessions.json:398]: Sort by similarity ratio

File: Packs/RDPCacheHunting/Layouts/layoutscontainer-RDP_Sessions.json [hunk 26]
```diff
@@ -0,0 +1,1311 @@
...
+                            }
+                        ],
+                        "h": 2,
+                        "i": "caseinfoid-field-changed-caseinfoid-06e49f40-0cdf-11ed-89d1-e5ac65360f00",
+                        "items": [],
+                        "maxW": 3,
+                        "minH": 1,
+                        "moved": false,
+                        "name": "Windows Path Indicators",
+                        "query": "type:\"Windows Path\"",
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Layouts/layoutscontainer-RDP_Sessions.json:263]: Change to 'Windows File Path'

File: Packs/RDPCacheHunting/Layouts/layoutscontainer-RDP_Sessions.json [hunk 27]
```diff
@@ -0,0 +1,1311 @@
...
+                        ],
+                        "description": "",
+                        "h": 2,
+                        "i": "caseinfoid-99427990-0d87-11ed-a87d-ffc9245ec67b",
+                        "items": [],
+                        "maxW": 3,
+                        "minH": 1,
+                        "moved": false,
+                        "name": "Executables Indicators",
+                        "query": "type:\"Executables\"",
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Layouts/layoutscontainer-RDP_Sessions.json:329]: Change to 'Windows Executables'

File: Packs/RDPCacheHunting/Layouts/layoutscontainer-RDP_Sessions.json [hunk 28]
```diff
@@ -0,0 +1,1311 @@
...
+                            }
+                        ],
+                        "h": 3,
+                        "i": "caseinfoid-8e64f340-0d87-11ed-a87d-ffc9245ec67b",
+                        "items": [],
+                        "maxW": 3,
+                        "minH": 1,
+                        "moved": false,
+                        "name": "Date and Time Indicators",
+                        "query": "type:\"Windows Date\u0026Time\"",
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Layouts/layoutscontainer-RDP_Sessions.json:304]: Change to 'Windows Date and Time'

File: Packs/RDPCacheHunting/Layouts/layoutscontainer-RDP_Sessions.json [hunk 29]
```diff
@@ -0,0 +1,1311 @@
...
+                        "type": "invTimeline",
+                        "w": 3,
+                        "x": 0,
+                        "y": 0
+                    }
+                ],
+                "type": "custom"
+            },
+            {
+                "id": "evidenceBoard",
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Layouts/layoutscontainer-RDP_Sessions.json:474]: Add the cache files and images as evidences.

File: Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt_README.md
Comment (melamedbn) [inline Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt_README.md:1]: re-generate docs

File: Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml [hunk 30]
```diff
@@ -0,0 +1,1900 @@
...
+      description:
+        simple: RDP Bitmap Cache OCR Output
+      customfields: {}
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 2
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "58":
```
Comment (melamedbn) [inline Packs/RDPCacheHunting/Playbooks/playbook-RDP_Bitmap_Cache_-_Detect_and_Hunt.yml:1720]: Missing "mark as evidence".

**Review by eyalpalo — APPROVED** [review]

**Comment by ShirleyDenkberg** [comment]
@efelmandar 
Need readmes for:
SetIndicatortype
SetRDPOverallScore

@melamedbn, @eyalpalo, @altmannyarden, @esharf Doc review completed.

**Pull request merged**